lua/weapons/weapon_dodgeball/shared.lua

```diff
@@ -4,63 +4,139 @@ SWEP.PrintName = "Dodgeball"
 SWEP.Slot = 0
 SWEP.ViewModel = Model("models/pyroteknik/dodgeball.mdl")
 SWEP.WorldModel = Model("models/pyroteknik/dodgeball.mdl")
+SWEP.Primary.Automatic = true
+SWEP.Secondary.Automatic = true
+SWEP.Primary.Ammo = "none"
+SWEP.Secondary.Ammo = "none" --you should really leave these in here so that people don't receive pistol ammo when they pick it up
+SWEP.Spawnable = true
+SWEP.Category = "Swamp Cinema" --todo remove these later i just need  this for testing 
 local outie = 64
 local innie = 28
 
+function SWEP:SetupDataTables()
+    self:NetworkVar("Int", 0, "ThrowState")
+    self:NetworkVar("Int", 1, "ThrowPower")
+    self:NetworkVar("Bool", 0, "Throwing")
+    self:NetworkVar("Float", 0, "StateTime")
+    self:NetworkVar("Entity", 0, "ThrownBall")
```

Comment:
This structure surely has duplicate data. Why do you need "Throwing" and "Throwstate" at the same time, can't you just use one value of Throwstate to indicate not thowing? you can even define GetThrowing to internally use the ThrowState

The reason to change this isn't a marginal performance improvement, it's because it is more of a single-source-of-truth which is a core principle in software engineering to reduce bugs. With the current setup there might be a bug where Throwing gets set but not ThrowState and causes some kind of weird issue.

Reply:
Right. I will look at this very soon